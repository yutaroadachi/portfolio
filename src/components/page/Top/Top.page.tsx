import React, { ComponentProps } from 'react'
import { Layout } from 'src/components/ui/Layout'
import { MetaTag } from 'src/components/ui/MetaTag'
import { Top } from './Top'

export type TopPageProps = ComponentProps<typeof Top>

export const TopPage = ({}: TopPageProps) => {
  return (
    <Layout pt={0}>
      <MetaTag
        title="トップ"
        description="Yutaro Adachi(フロントエンドエンジニア)のポートフォリオサイトです。"
      />
      <Top />
    </Layout>
  )
}
