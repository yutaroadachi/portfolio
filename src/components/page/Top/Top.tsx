import { ChevronDownIcon } from '@chakra-ui/icons'
import {
  Box,
  chakra,
  Icon,
  Img,
  LinkBox,
  LinkOverlay,
  SimpleGrid,
  Stack,
  VStack,
} from '@chakra-ui/react'
import Head from 'next/head'
import NextLink from 'next/link'
import React from 'react'
import { Biography } from 'src/components/domain/Biography'
import { LINKS } from 'src/constants/links'

export type TopProps = {}

export const Top = ({}: TopProps) => {
  return (
    <Stack spacing={8}>
      <Hero />
      <AboutMe />
      <Links />
    </Stack>
  )
}

const Hero = () => {
  const HEADER_HEIGHT = 64
  const ADRESS_BAR_HEIGHT = 64

  return (
    <>
      <Head>
        <link rel="preload" href="/assets/top-page/hero.png" as="image" />
      </Head>
      <VStack
        spacing={4}
        justify="center"
        align="center"
        minH={{
          base: `calc(100vh - ${HEADER_HEIGHT + ADRESS_BAR_HEIGHT}px)`,
          lg: `calc(100vh - ${HEADER_HEIGHT}px)`,
        }}
      >
        <chakra.h1 id="hero" textStyle="h1" color="main" textAlign="center">
          adachi&apos;s
          <br />
          portfolio site
        </chakra.h1>
        <Img
          src="/assets/top-page/hero.png"
          htmlWidth="313px"
          htmlHeight="333px"
          alt=""
        />
        <VStack spacing={0} color="main">
          <ChevronDownIcon boxSize="48px" />
          <chakra.p fontSize="24px" fontWeight="700">
            Scroll
          </chakra.p>
        </VStack>
      </VStack>
    </>
  )
}

const AboutMe = () => {
  return (
    <Box>
      <chakra.h2 id="about-me" textStyle="h2WithTreat" color="main" mb={8}>
        私について
      </chakra.h2>
      <Biography />
    </Box>
  )
}

const Links = () => {
  return (
    <Box w="full">
      <chakra.h2 id="links" textStyle="h2WithTreat" color="main" mb={8}>
        リンク
      </chakra.h2>
      <SimpleGrid columns={{ base: 2, md: 4 }} gap={4}>
        {LINKS.map((link) => {
          return (
            <LinkBox layerStyle="card" p={4} key={link.title}>
              <VStack spacing={1}>
                <Icon boxSize="48px" as={link.icon} />
                <NextLink href={link.href} passHref>
                  <LinkOverlay
                    fontWeight="700"
                    data-test-id={`top-to-${link.key}`}
                  >
                    {link.title}
                  </LinkOverlay>
                </NextLink>
              </VStack>
            </LinkBox>
          )
        })}
      </SimpleGrid>
    </Box>
  )
}
