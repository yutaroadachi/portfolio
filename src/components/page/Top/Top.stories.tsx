import { Meta, Story } from '@storybook/react'
import React from 'react'
import { Top, TopProps } from './Top'

export default {
  title: 'page/Top',
  component: Top,
} as Meta<TopProps>

type Template = Story<TopProps>

const Template: Template = (args) => <Top {...args} />

export const Default: Template = Template.bind({})
