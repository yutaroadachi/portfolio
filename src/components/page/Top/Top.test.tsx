import React from 'react'
import { render } from 'src/__tests__/utils'
import { Top } from './Top'

test('Top', () => {
  const { container } = render(<Top />)
  expect(container).toMatchSnapshot()
})
