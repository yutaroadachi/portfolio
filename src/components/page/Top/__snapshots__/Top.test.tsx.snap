// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top 1`] = `
<div>
  <div
    class="chakra-stack css-aajx3r"
  >
    <div
      class="chakra-stack css-1fzzft9"
    >
      <h1
        class="css-1nukpjn"
        id="hero"
      >
        adachi's
        <br />
        portfolio site
      </h1>
      <img
        alt=""
        class="chakra-image css-0"
        height="333px"
        src="/assets/top-page/hero.png"
        width="313px"
      />
      <div
        class="chakra-stack css-ujpg6b"
      >
        <svg
          class="chakra-icon css-l1hgsr"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
        <p
          class="css-fyih5c"
        >
          Scroll
        </p>
      </div>
    </div>
    <div
      class="css-0"
    >
      <h2
        class="css-jjss8v"
        id="about-me"
      >
        私について
      </h2>
      <div
        class="chakra-stack css-cx70wh"
      >
        <div
          class="chakra-stack css-3c05cu"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27128%27%20height=%27128%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Yutaro Adachiのプロフィール画像"
              class="circle-image"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="chakra-stack css-igigla"
          >
            <h2
              class="css-553wiy"
            >
              Yutaro Adachi
            </h2>
            <p
              class="css-1vf9ext"
            >
              フロントエンドエンジニア
            </p>
          </div>
          <div
            class="chakra-stack css-a9v878"
          >
            <a
              class="chakra-link css-f4h6uy"
              href="https://github.com/yutaroadachi"
              rel="noopener"
              target="_blank"
            >
              <svg
                class="chakra-icon css-1fmr1qf"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
            </a>
            <a
              class="chakra-link css-f4h6uy"
              href="https://twitter.com/a_da_chi_"
              rel="noopener"
              target="_blank"
            >
              <svg
                class="chakra-icon css-1fmr1qf"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                />
              </svg>
            </a>
          </div>
        </div>
        <p
          class="css-0"
        >
          大学で経済学を学んだのち、大手SIerでメンバーおよびチームリーダーとして客先常駐でのシステム開発を経験しました。この経験を通して、自分でもコードを書けるようになる必要があると感じ、エンジニアにキャリアチェンジすることを決意しました。
          <br />
          その後スタートアップでエンジニアとしてフロントエンド開発からバックエンド開発まで幅広く経験してきましたが、特にReactを中心としたフロントエンド開発が得意です。また要件定義や設計といった上流工程の経験もしてきたため、一人称で開発に携わることができると自負しております。
        </p>
      </div>
    </div>
    <div
      class="css-1ki54i"
    >
      <h2
        class="css-jjss8v"
        id="links"
      >
        リンク
      </h2>
      <div
        class="css-1iv1hce"
      >
        <div
          class="chakra-linkbox css-ujffn4"
        >
          <div
            class="chakra-stack css-126kq63"
          >
            <svg
              class="chakra-icon css-1ttq8fd"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M418.2 177.2c-5.4-1.8-10.8-3.5-16.2-5.1.9-3.7 1.7-7.4 2.5-11.1 12.3-59.6 4.2-107.5-23.1-123.3-26.3-15.1-69.2.6-112.6 38.4-4.3 3.7-8.5 7.6-12.5 11.5-2.7-2.6-5.5-5.2-8.3-7.7-45.5-40.4-91.1-57.4-118.4-41.5-26.2 15.2-34 60.3-23 116.7 1.1 5.6 2.3 11.1 3.7 16.7-6.4 1.8-12.7 3.8-18.6 5.9C38.3 196.2 0 225.4 0 255.6c0 31.2 40.8 62.5 96.3 81.5 4.5 1.5 9 3 13.6 4.3-1.5 6-2.8 11.9-4 18-10.5 55.5-2.3 99.5 23.9 114.6 27 15.6 72.4-.4 116.6-39.1 3.5-3.1 7-6.3 10.5-9.7 4.4 4.3 9 8.4 13.6 12.4 42.8 36.8 85.1 51.7 111.2 36.6 27-15.6 35.8-62.9 24.4-120.5-.9-4.4-1.9-8.9-3-13.5 3.2-.9 6.3-1.9 9.4-2.9 57.7-19.1 99.5-50 99.5-81.7 0-30.3-39.4-59.7-93.8-78.4zM282.9 92.3c37.2-32.4 71.9-45.1 87.7-36 16.9 9.7 23.4 48.9 12.8 100.4-.7 3.4-1.4 6.7-2.3 10-22.2-5-44.7-8.6-67.3-10.6-13-18.6-27.2-36.4-42.6-53.1 3.9-3.7 7.7-7.2 11.7-10.7zM167.2 307.5c5.1 8.7 10.3 17.4 15.8 25.9-15.6-1.7-31.1-4.2-46.4-7.5 4.4-14.4 9.9-29.3 16.3-44.5 4.6 8.8 9.3 17.5 14.3 26.1zm-30.3-120.3c14.4-3.2 29.7-5.8 45.6-7.8-5.3 8.3-10.5 16.8-15.4 25.4-4.9 8.5-9.7 17.2-14.2 26-6.3-14.9-11.6-29.5-16-43.6zm27.4 68.9c6.6-13.8 13.8-27.3 21.4-40.6s15.8-26.2 24.4-38.9c15-1.1 30.3-1.7 45.9-1.7s31 .6 45.9 1.7c8.5 12.6 16.6 25.5 24.3 38.7s14.9 26.7 21.7 40.4c-6.7 13.8-13.9 27.4-21.6 40.8-7.6 13.3-15.7 26.2-24.2 39-14.9 1.1-30.4 1.6-46.1 1.6s-30.9-.5-45.6-1.4c-8.7-12.7-16.9-25.7-24.6-39s-14.8-26.8-21.5-40.6zm180.6 51.2c5.1-8.8 9.9-17.7 14.6-26.7 6.4 14.5 12 29.2 16.9 44.3-15.5 3.5-31.2 6.2-47 8 5.4-8.4 10.5-17 15.5-25.6zm14.4-76.5c-4.7-8.8-9.5-17.6-14.5-26.2-4.9-8.5-10-16.9-15.3-25.2 16.1 2 31.5 4.7 45.9 8-4.6 14.8-10 29.2-16.1 43.4zM256.2 118.3c10.5 11.4 20.4 23.4 29.6 35.8-19.8-.9-39.7-.9-59.5 0 9.8-12.9 19.9-24.9 29.9-35.8zM140.2 57c16.8-9.8 54.1 4.2 93.4 39 2.5 2.2 5 4.6 7.6 7-15.5 16.7-29.8 34.5-42.9 53.1-22.6 2-45 5.5-67.2 10.4-1.3-5.1-2.4-10.3-3.5-15.5-9.4-48.4-3.2-84.9 12.6-94zm-24.5 263.6c-4.2-1.2-8.3-2.5-12.4-3.9-21.3-6.7-45.5-17.3-63-31.2-10.1-7-16.9-17.8-18.8-29.9 0-18.3 31.6-41.7 77.2-57.6 5.7-2 11.5-3.8 17.3-5.5 6.8 21.7 15 43 24.5 63.6-9.6 20.9-17.9 42.5-24.8 64.5zm116.6 98c-16.5 15.1-35.6 27.1-56.4 35.3-11.1 5.3-23.9 5.8-35.3 1.3-15.9-9.2-22.5-44.5-13.5-92 1.1-5.6 2.3-11.2 3.7-16.7 22.4 4.8 45 8.1 67.9 9.8 13.2 18.7 27.7 36.6 43.2 53.4-3.2 3.1-6.4 6.1-9.6 8.9zm24.5-24.3c-10.2-11-20.4-23.2-30.3-36.3 9.6.4 19.5.6 29.5.6 10.3 0 20.4-.2 30.4-.7-9.2 12.7-19.1 24.8-29.6 36.4zm130.7 30c-.9 12.2-6.9 23.6-16.5 31.3-15.9 9.2-49.8-2.8-86.4-34.2-4.2-3.6-8.4-7.5-12.7-11.5 15.3-16.9 29.4-34.8 42.2-53.6 22.9-1.9 45.7-5.4 68.2-10.5 1 4.1 1.9 8.2 2.7 12.2 4.9 21.6 5.7 44.1 2.5 66.3zm18.2-107.5c-2.8.9-5.6 1.8-8.5 2.6-7-21.8-15.6-43.1-25.5-63.8 9.6-20.4 17.7-41.4 24.5-62.9 5.2 1.5 10.2 3.1 15 4.7 46.6 16 79.3 39.8 79.3 58 0 19.6-34.9 44.9-84.8 61.4zm-149.7-15c25.3 0 45.8-20.5 45.8-45.8s-20.5-45.8-45.8-45.8c-25.3 0-45.8 20.5-45.8 45.8s20.5 45.8 45.8 45.8z"
              />
            </svg>
            <a
              class="chakra-linkbox__overlay css-19sbzu9"
              data-test-id="top-to-skills"
              href="/skills"
            >
              スキル
            </a>
          </div>
        </div>
        <div
          class="chakra-linkbox css-ujffn4"
        >
          <div
            class="chakra-stack css-126kq63"
          >
            <svg
              class="chakra-icon css-1ttq8fd"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 221.25V416a48 48 0 01-48 48H144a48 48 0 01-48-48V96a48 48 0 0148-48h98.75a32 32 0 0122.62 9.37l141.26 141.26a32 32 0 019.37 22.62z"
                fill="none"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <path
                d="M256 56v120a32 32 0 0032 32h120m-232 80h160m-160 80h160"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            <a
              class="chakra-linkbox__overlay css-19sbzu9"
              data-test-id="top-to-resume"
              href="/resume"
            >
              履歴書
            </a>
          </div>
        </div>
        <div
          class="chakra-linkbox css-ujffn4"
        >
          <div
            class="chakra-stack css-126kq63"
          >
            <svg
              class="chakra-icon css-1ttq8fd"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5,16 C3.343,16 2,17.343 2,19 C2,20.657 3.343,22 5,22 C6.657,22 8,20.657 8,19 C8,17.343 6.657,16 5,16 L5,16 L5,16 Z M5,1 C14.925,1 23,9.075 23,19 L23,19 M18,19 C18,11.832 12.168,6 5,6 M13,19 C13,14.589 9.411,11 5,11 M2,11 L2,19 L2,19"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
            <a
              class="chakra-linkbox__overlay css-19sbzu9"
              data-test-id="top-to-blog"
              href="/blog"
            >
              ブログ
            </a>
          </div>
        </div>
        <div
          class="chakra-linkbox css-ujffn4"
        >
          <div
            class="chakra-stack css-126kq63"
          >
            <svg
              class="chakra-icon css-1ttq8fd"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="320"
                rx="40"
                ry="40"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
                width="416"
                x="48"
                y="96"
              />
              <path
                d="M112 160l144 112 144-112"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            <a
              class="chakra-linkbox__overlay css-19sbzu9"
              data-test-id="top-to-contact"
              href="https://docs.google.com/forms/d/e/1FAIpQLSdSMQFzMIm2oWZ9fk_r1RGIUl_gaf3r6ouHVK_Ysn2FERGxVA/viewform?usp=sf_link"
              rel="noopener noreferrer"
              target="_blank"
            >
              お問い合わせ
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
